docs/acls.md

```diff
@@ -0,0 +1,288 @@
+registering it is allowed to do it.
+
+Here are the ACL's to implement the same permissions as above:
+
+```json
+{
+    // groups are simpler and only list the namespaces name
+    "groups": {
+        "group:boss": ["boss"],
+        "group:dev": ["dev1","dev2"],
+        "group:admin": ["admin1"],
+        "group:intern": ["intern1"],
+    },
+    "tagOwners": {
+        // the administrators can add servers in production
+        "tag:prod-databases": ["group:admin"],
+        "tag:prod-app-servers": ["group:admin"],
+
+        // the boss can tag any server as internal
+        "tag:internal": ["group:boss"],
+
+        // dev can add servers for dev purposes as well as admins
+        "tag:dev-databases": ["group:admin","group:dev"],
+        "tag:dev-app-servers": ["group:admin", "group:dev"],
+
+        // interns cannot add servers
+    },
+    "acls": [
+        // boss have access to all servers
+        {"action":"accept",
+         "users":["group:boss"],
+         "ports":[
+            "tag:prod-databases:*",
+            "tag:prod-app-servers:*",
+            "tag:internal:*",
+            "tag:dev-databases:*",
+            "tag:dev-app-servers:*",
+            ]
+        },
+
+        // admin have only access to administrative ports of the servers
+        {"action":"accept",
+         "users":["group:admin"],
+         "ports":[
+            "tag:prod-databases:22",
+            "tag:prod-app-servers:22",
+            "tag:internal:22",
+            "tag:dev-databases:22",
+            "tag:dev-app-servers:22",
+            ]
+        },
+
+        {"action":"accept", "users":["group:dev"], "ports":[
+            "tag:dev-databases:*",
+            "tag:dev-app-servers:*",
+            "tag:prod-app-servers:80,443",
+            ]
+        },
+
+        // servers should be able to talk to database. Database should not be able to initiate connections to server
+        {"action":"accept", "users":["tag:dev-app-servers"], "ports":["tag:dev-databases:5432"]},
+        {"action":"accept", "users":["tag:prod-app-servers"], "ports":["tag:prod-databases:5432"]},
+
+        // interns have access to dev-app-servers only in reading mode
+        {"action":"accept", "users":["group:intern"], "ports":["tag:dev-app-servers:80,443"]},
+
+        // we still have to allow internal namespaces communications since nothing guarantees that each user have their own namespaces. This could be talked over.
+        {"action":"accept", "users":["boss"], "ports":["boss:*"]},
+        {"action":"accept", "users":["dev1"], "ports":["dev1:*"]},
+        {"action":"accept", "users":["dev2"], "ports":["dev2:*"]},
+        {"action":"accept", "users":["admin1"], "ports":["admin1:*"]},
+        {"action":"accept", "users":["intern1"], "ports":["intern1:*"]},
+    ]
+}
+```
+
+With this implementation, the sharing step is not necessary. Maintenance cost of the ACL file is lower and less tedious (no need to map hostname and IP's into it).
```

Comment:
I like this, it would also make the OIDC implementation a lot more sensible (one login -> one namespace).

Reply:
No need hostname and IP mapping is great. Generally this feature should be preserved.